.travis.yml

```diff
@@ -1,12 +1,39 @@
+sudo: required
+# Trusty distribution is much faster when sudo is required
+dist: trusty
+
+services:
+  - docker
+
 language: go
 go:
   - 1.5
   - 1.6
+
+# In the Travis VM-based build environment, IPv6 networking is not
+# enabled by default. The sysctl operations below enable IPv6.
+# IPv6 is needed by some of the CoreDNS test cases. The VM environment
+# is needed to have sudo in the test environment. Sudo is needed to have
+# docker in the test environment. Docker is needed to launch a kubernetes
+# instance in the test environment.
+# (Dependencies are fun! :) )
+before_install:
+  - cat /proc/net/if_inet6
+  - uname -a
+  - sudo bash -c 'if [ `cat /proc/net/if_inet6 | wc -l` = "0" ]; then echo "Enabling IPv6" ; sysctl net.ipv6.conf.all.disable_ipv6=0 ; sysctl net.ipv6.conf.default.disable_ipv6=0 ; sysctl net.ipv6.conf.lo.disable_ipv6=0 ; fi'
+  - cat /proc/net/if_inet6
+  - env
+
```

Comment:
Nothing official.

I contacted Travis support and they claimed that IPv6 works in the VM environment but not in the container environment. I provided https://github.com/mrichmon/validate-travis-network to illustrate the issue. Two weeks later and no movement by Travis.
Also provided an update to their docs to correctly describe the behavior in https://github.com/travis-ci/docs-travis-ci-com/pull/662

With the conditional line in the script, if IPv6 is already enabled then nothing is altered. This should be fairly future-proof.